src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1323,6 +1330,71 @@ private void SetTotalDifficulty(BlockHeader header)
 
         public event EventHandler<BlockEventArgs> NewHeadBlock;
 
+        public void DeleteChainSlice(in long startNumber, in long endNumber)
+        {
+            if (endNumber - startNumber < 0)
+            {
+                throw new InvalidOperationException($"Start number must be equal or greater end number.");
+            }
+            
+            if (endNumber - startNumber > 1000)
+            {
+                throw new InvalidOperationException($"Cannot delete that many blocks at once (start: {startNumber}, end {endNumber}).");
+            }
+            
+            if (endNumber < BestKnownNumber)
+            {
+                throw new InvalidOperationException("End number has to be at least equal to the best known number when deleting a slice");
+            }
+
+            if (startNumber < 1)
+            {
+                throw new InvalidOperationException("Start number must be strictly greater than 0");
+            }
+
+            Block newHeadBlock = null;
+            
+            // we are running these checks before all the deletes
+            if (Head.Number >= startNumber)
+            {
+                // greater than zero so will not fail
+                ChainLevelInfo chainLevelInfo = _chainLevelInfoRepository.LoadLevel(startNumber - 1);
+
+                // there may be no canonical block marked on this level - then we just hack to genesis
+                Keccak newHeadHash = chainLevelInfo.HasBlockOnMainChain ? chainLevelInfo.BlockInfos[0].BlockHash : Genesis.Hash;
+                newHeadBlock = FindBlock(newHeadHash, BlockTreeLookupOptions.None);
+            }
+
+            using (_chainLevelInfoRepository.StartBatch())
+            {
+                for (long i = endNumber; i >= startNumber; i--)
+                {
+                    ChainLevelInfo chainLevelInfo = _chainLevelInfoRepository.LoadLevel(startNumber);
```

Comment:
I think you meant to use i instead of startNumber in this loop for LoadLevel and Delete

Reply:
thanks, fixed, and added a test that discovers it